### PR TITLE
Move the graphql queries that belong in core

### LINKED ISSUE #416: Separate api from implementation
This project will contain only the api, spi, api-helper, and graphql packages. This defines the interfaces and provides common methods for implementing them, without specifying a triggering mechanism or full implementation of the Software Delivery Machine.

The rest of the code will move to sdm-core.

- [X] disengage these packages from all others
- [X] move tests into the corresponding packages
- [X] break out into new project
- [x] update the imports in downstream projects
    - [x] packs
    - [x] concrete SDMs

### PR DESCRIPTION
**Pull request by jessitron**
Fixes #416 

... in combination with the new sdm-core repository (master branch).

This is a very breaking change; dependencies and imports are quite different in all downstream projects. However, most breakages were handled by an editor (not currently deployed). There are branches in all downstream projects for this.

### CODE REVIEW

File: src/api-helper/log/addressChannelsProgressLog.ts [hunk 1]
```diff
@@ -15,18 +15,18 @@
...
- * with the current model element (such a repo)
+ * Stream the ProgressLog output to any channels associated
+ * with the current model element (such a repo) in Slack or elswhere.
  * @param name name for the log. Should relate to the activity we're logging
  * @param {HasChannels} hasChannels
  * @param {HandlerContext} ctx
  * @return {ProgressLog}
  */
-export function slackProgressLog(name: string, hasChannels: HasChannels, ctx: HandlerContext): ProgressLog {
+export function addressChannelsProgressLog(name: string, hasChannels: HasChannels, ctx: HandlerContext): ProgressLog {
```
Comment (johnsonr) [inline src/api-helper/log/addressChannelsProgressLog.ts]: Good change

File: src/api-helper/misc/reportFailureInterpretation.ts [hunk 2]
```diff
@@ -20,12 +20,12 @@ import * as slack from "@atomist/slack-messages/SlackMessages";
 import { AddressChannels } from "../../api/context/addressChannels";
 import { InterpretedLog } from "../../spi/log/InterpretedLog";
 
-export async function reportFailureInterpretationToLinkedChannels(stepName: string,
-                                                                  interpretation: InterpretedLog | undefined,
-                                                                  fullLog: { url?: string, log: string },
-                                                                  id: RemoteRepoRef,
-                                                                  addressChannels: AddressChannels,
-                                                                  retryButton?: slack.Action) {
+export async function reportFailureInterpretation(stepName: string,
```
Comment (johnsonr) [inline src/api-helper/misc/reportFailureInterpretation.ts]: Nice

File: src/api-helper/misc/sha.ts [hunk 3]
```diff
@@ -16,10 +16,6 @@
 
 import * as jsSHA from "jssha";
 
-export function isValidSHA1(s: string): boolean {
-    return s.match(/[a-fA-F0-9]{40}/) != null;
-}
-
```
Comment (johnsonr) [inline src/api-helper/misc/sha.ts]: Did this move somewhere?

File: src/api/registration/EditorRegistration.ts [hunk 4]
```diff
@@ -41,8 +45,16 @@ export interface EditorRegistration<PARAMS = EmptyParameters> extends Partial<Ed
     /**
      * Should this be a dry run editor: That is,
      * should it wait for the build result to determine whether to raise a pull request
-     * or issue. Default is no.
+     * or issue. Default is editorCommand, which creates a branch and a PR.
      */
-    dryRun?: boolean;
+    editorCommandFactory?: EditorCommandFactory<PARAMS>;
```
Comment (johnsonr) [inline src/api/registration/EditorRegistration.ts]: Not sure the comment is quite right now. Isn't this a more general mechanism?

File: src/api/registration/EditorRegistration.ts [hunk 5]
```diff
@@ -41,8 +45,16 @@ export interface EditorRegistration<PARAMS = EmptyParameters> extends Partial<Ed
...
 
 }
+
+export type EditorCommandFactory<PARAMS> = (
```
Comment (johnsonr) [inline src/api/registration/EditorRegistration.ts]: Would be great to have a comment on this type

File: src/api-helper/log/addressChannelsProgressLog.ts (hunk 1 above)
Comment (jessitron) [reply src/api-helper/log/addressChannelsProgressLog.ts]: pretty sure that was yours ;-)

File: src/api/registration/EditorRegistration.ts (hunk 4 above)
Comment (jessitron) [reply src/api/registration/EditorRegistration.ts]: yes good point

File: src/api/registration/EditorRegistration.ts (hunk 5 above)
Comment (jessitron) [reply src/api/registration/EditorRegistration.ts]: ok but I don't have a deep understanding of the arguments; I don't think it's really a clean abstraction, but it is a step in the direction we needed.

File: src/api-helper/misc/sha.ts (hunk 3 above)
Comment (jessitron) [reply src/api-helper/misc/sha.ts]: yes, to the one test where it was used.

**Pull request merged**